src/jsonrpc.js

```diff
@@ -0,0 +1,87 @@
+import logger from './logger';
+
+/**
+ * This is promisify function that convert function with callback to promise
+ * @param {*} func 
+ */
+export const toPromise = func => (...args) =>
+  new Promise((resolve, reject) =>
+    func(...args, (error, result) => (error ? reject(new Error(error.message)) : resolve(result)))
+  );
+
+
+export class JsonRpc {
+  /**
+   * Initialize class local variables
+   * @param {*} addresses 
+   * @param {*} fromBlock 
+   * @param {*} toBlock 
+   * @param {*} callback 
+   */
+  constructor(addresses, fromBlock, toBlock, web3, callback = null) {
+    this.addresses = addresses.map(address => address.toLowerCase());
+    this.fromBlock = fromBlock;
+    this.toBlock = toBlock;
+    this.web3Instance = web3.eth;
+    this.callback = callback;
+  }
+
+  /**
+   * 
+   * This will return the final data structure 
+   * for the transaction resopnse
+   * @param {*} tx 
+   * @param {*} receipt  
+   * @return object
+   */
+  _getTransaction(tx, receipt) {
+    return { ...tx, ...receipt };
+  }
+
+  /**
+   * Async function that get the transaction and transaction receipt then get the 
+   * logs out of the receipt transaction then execute the callback function
+   * @param {*} txn
+   */
+  async _scanTransaction(txn) {
+    const txnReceipts = await toPromise(this.web3Instance.getTransactionReceipt)(txn.hash);
+    if ((txn.to && this.addresses.indexOf(txn.to.toLowerCase()) >= 0) ||
+      (txn.from && this.addresses.indexOf(txn.from.toLowerCase()) >= 0)) {
+      const txnNormal = await toPromise(this.web3Instance.getTransaction)(txn.hash);
+      const transactionResult = this._getTransaction(txnNormal, txnReceipts);
+      if (this.callback) { this.callback(transactionResult, txnReceipts.logs); }
+    }
+  }
+
+  /**
+   * This function handling the tranactions that exists in one block and 
```

Comment:
This function handles the transactions that exist in one block and puts them into an array of promises, then executes them.